modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/js/components/OrderableTable.tsx

```diff
@@ -0,0 +1,322 @@
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayButton, {ClayButtonWithIcon} from '@clayui/button';
+import ClayEmptyState from '@clayui/empty-state';
+import ClayLayout from '@clayui/layout';
+import ClayTable from '@clayui/table';
+import classNames from 'classnames';
+import {ManagementToolbar} from 'frontend-js-components-web';
+import fuzzy from 'fuzzy';
+import React, {useRef, useState} from 'react';
+import {DndProvider, useDrag, useDrop} from 'react-dnd';
+import {HTML5Backend} from 'react-dnd-html5-backend';
+
+import Search from './Search';
+
+import '../../css/OrderableTable.scss';
+
+const FUZZY_OPTIONS = {
+	post: '</strong>',
+	pre: '<strong>',
+};
+
+interface OrderableTableRowInterface {
+	fields: Array<{
+		label: string;
+		name: string;
+	}>;
+	index: number;
+	item: any;
+	onOrderChange: Function;
+	query: string;
+}
+
+const OrderableTableRow = ({
+	fields,
+	index,
+	item,
+	onOrderChange,
+	query,
+}: OrderableTableRowInterface) => {
+	const ref = useRef<HTMLTableRowElement>(null);
+
+	const [{isDragging}, dragRef] = useDrag({
+		collect: (monitor) => ({
+			isDragging: monitor.isDragging(),
+		}),
+		item: {
+			index,
+			type: 'FIELD',
+		},
+	});
+
+	const [, dropRef] = useDrop({
+		accept: 'FIELD',
+		hover(item: {index: number; type: string}, monitor) {
+			if (!ref.current || !onOrderChange) {
+				return;
+			}
+
+			const draggedIndex = item.index;
+			const targetIndex = index;
+
+			if (draggedIndex === targetIndex) {
+				return;
+			}
+
+			const targetSize = ref.current.getBoundingClientRect();
+			const targetCenter = (targetSize.bottom - targetSize.top) / 2;
+
+			const draggedOffset: {
+				x: number;
+				y: number;
+			} | null = monitor.getClientOffset();
+
+			if (!draggedOffset) {
+				return;
+			}
+
+			const draggedTop = draggedOffset.y - targetSize.top;
+
+			if (
+				(draggedIndex < targetIndex && draggedTop < targetCenter) ||
+				(draggedIndex > targetIndex && draggedTop > targetCenter)
+			) {
+				return;
+			}
+
+			onOrderChange({draggedIndex, targetIndex});
+
+			item.index = targetIndex;
+		},
+	});
+
+	dragRef(dropRef(ref));
+
+	return (
+		<ClayTable.Row
+			className={classNames('orderable-table-row', {
+				dragging: isDragging,
+			})}
+			key={index}
+			ref={ref}
+		>
+			<ClayTable.Cell className="drag-handle-cell">
+				<ClayButtonWithIcon
+					aria-label={Liferay.Util.sub(
+						Liferay.Language.get('drag-x'),
+						item.label || Liferay.Language.get('item')
+					)}
+					displayType={null}
+					symbol="drag"
+				/>
+			</ClayTable.Cell>
+
+			{fields.map((field) => {
+				const itemFieldValue = String(item[field.name]);
+
+				const fuzzyMatch = fuzzy.match(
+					query,
+					itemFieldValue,
+					FUZZY_OPTIONS
+				);
+
+				return (
+					<ClayTable.Cell key={field.name}>
```

Comment:
Are field names unique? If they aren't you might have problems with drag and drop.